src/Microsoft.ML.Data/Transforms/SelectColumnsTransform.cs

```diff
@@ -0,0 +1,632 @@
+                int outputIdx = 0;
+
+                for (int colIdx = 0; colIdx < columnCount; ++colIdx)
+                {
+                    if (!keepHidden && inputSchema.IsHidden(colIdx))
+                        continue;
+
+                    var columnName = inputSchema[colIdx].Name;
+                    if (selectedColumns.Contains(columnName))
+                    {
+                        outputToInputMapping.Add(colIdx);
+                        outputIdx++;
+                    }
+                }
+
+                return outputToInputMapping.ToArray();
+            }
+
+            private static Schema GenerateOutputSchema(IEnumerable<string> selectedColumns,
+                                                        bool keepHidden,
+                                                        Schema inputSchema)
+            {
+                var schemaColumns = new List<Schema.Column>();
+                var columnCount = inputSchema.ColumnCount;
+                for (int colIdx = 0; colIdx < columnCount; ++colIdx)
+                {
+                    if (!keepHidden && inputSchema.IsHidden(colIdx))
+                        continue;
+
+                    var column = inputSchema[colIdx];
+                    if (selectedColumns.Contains(column.Name))
+                    {
+                        schemaColumns.Add(column);
+                    }
+                }
+                return new Schema(schemaColumns);
+            }
+        }
+
+        private sealed class Row : IRow
+        {
+            private readonly Mapper _mapper;
+            private readonly IRow _input;
+            public Row(IRow input, Mapper mapper, Schema outputSchema)
+            {
+                _mapper = mapper;
+                _input = input;
+            }
+
+            public long Position => _input.Position;
+
+            public long Batch => _input.Batch;
+
+            public ISchema Schema => _mapper.Schema;
+
+            public ValueGetter<TValue> GetGetter<TValue>(int col)
+            {
+                int index = _mapper[col];
+                return _input.GetGetter<TValue>(index);
+            }
+
+            public ValueGetter<UInt128> GetIdGetter()
+                => _input.GetIdGetter();
+
+            public bool IsColumnActive(int col)
+            {
+                return true;
+            }
+        }
+
+        private sealed class SelectColumnsDataTransform : IDataTransform, IRowToRowMapper
+        {
+            private readonly Mapper _mapper;
+            private readonly IHostEnvironment _host;
+            private readonly SelectColumnsTransform _transform;
+            public SelectColumnsDataTransform(IHostEnvironment host, SelectColumnsTransform transform, Mapper mapper, IDataView input)
+            {
+                _transform = transform;
+                _mapper = mapper;
+                Source = input;
+                _host = host;
```

Comment:
>_host = host [](start = 16, length = 12)

accept env object and make _host is IHost, and do _host = env.Register(nameof(SelectColumnsDataTransform));